engines/script/sandbox.go

```diff
@@ -0,0 +1,138 @@
+package scriptengine
+
+import (
+	"mime"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+)
+
+type sandbox struct {
+	engines.SandboxBase
+	context     *runtime.TaskContext
+	engine      *engine
+	cmd         *exec.Cmd
+	folder      runtime.TemporaryFolder
+	resolve     atomics.Once
+	resultSet   engines.ResultSet
+	resultError error
+	resultAbort error
+	log         *logrus.Entry
+	aborted     atomics.Bool
+	done        chan struct{}
+}
+
+func (s *sandbox) run() {
+	err := s.cmd.Wait()
+	success := err == nil
+
+	// if error wasn't because script exited non-zero, then we have a problem
+	if _, ok := err.(*exec.ExitError); err != nil && !ok {
+		s.log.Error("Script execution failed, error: ", err)
+	}
+
+	// Upload artifacts if not aborted
+	if !s.aborted.Get() {
+		err2 := s.uploadArtifacts()
+		if err2 != nil {
+			success = false
+			s.context.LogError("Failed to upload artifacts")
```

Comment:
No, I strongly disagree.
`err2` is an internal error.

---

So I see us as having two kinds of errors.
 1. errors that we blame the task author for
 2. internal errors... (all other errors)

If we are not 100% sure it's the task authors fault we have no business telling the task author what the internal error is.
In fact I view doing so as a security risk, error messages could contain headers with auth stuff etc... so we should **never** blindly log errors to task logs.